### PR TITLE
add extra sidecar container to dashboard in helm chart

### LINKED ISSUE #4155: allow sidecar in helm chart
## Feature Request

Because the dashboard is very important pod I need to add an OAuth sidecar to it but helm chart seems doesn't support any additional container. I think we can add it to helm easily. Thanks

### PR DESCRIPTION
**Pull request by jamshidi799**
## What problem does this PR solve?
This PR is related to https://github.com/chaos-mesh/chaos-mesh/issues/4155. It supports additional sidecar containers in the dashboard pod in the helm chart.
I can also add sidecar support to other deployments if necessary.

Close #4155 

## Related changes

- [ ] This change also requires further updates to the [website](https://github.com/chaos-mesh/website) (e.g. docs)
- [ ] This change also requires further updates to the `UI interface`

## Cherry-pick to release branches (optional)

> This PR should be cherry-picked to the following release branches:

- [ ] release-2.6
- [ ] release-2.5

### CHANGELOG

> Must include at least one of them.

- [x] I have updated the `CHANGELOG.md`
- [ ] I have labeled this PR with "no-need-update-changelog"

### Tests

> Must include at least one of them.

- [ ] Unit test
- [ ] E2E test
- [x] Manual test

### Side effects

- [ ] **Breaking backward compatibility**

## DCO

If you find the DCO check fails, please run commands like below (Depends on the actual situations. For example, if the failed commit isn't the most recent) to fix it:

```shell
git commit --amend --signoff
git push --force
```

### CODE REVIEW

**Comment by cwen0** [comment]
@jamshidi799 Can you update the change logs?

File: helm/chaos-mesh/values.yaml [hunk 1]
```diff
@@ -283,6 +283,8 @@ dashboard:
   affinity: {}
   # Deployment chaos-dashboard annotations
   podAnnotations: {}
+  # list of extra side car containers
```
Comment (g1eny0ung) [inline helm/chaos-mesh/values.yaml:286]:
```suggestion
  # List of extra sidecar containers
```

**Review by g1eny0ung — CHANGES_REQUESTED** [review]

**Review by g1eny0ung — APPROVED** [review]
/lgtm

Thanks for your contribution! ❤️

**Comment by jamshidi799** [comment]
> /lgtm
> 
> Thanks for your contribution! ❤️

Thank you too. I'm eager to do more.

**Pull request merged**